### PR TITLE
Added support for result source configuration by scope and name

### PR DESCRIPTION
**Pull request by pettersh**
As discussed with @wobba I've ported the result source name feature from v3 to v4

How to test the PR:

1. Create a custom search result source "Source1" for the Site (SPWeb) using a unique query template, for example `{?{searchTerms} SOMETHING#1}`
1. Create a custom search result source "Source2" for the Site collection (SPWeb) using a unique query template, for example `{?{searchTerms} SOMETHING#2}`
1. Create a custom search result source "Source3" for the Tenant (SPSiteSubscription) using a unique query template, for example `{?{searchTerms} SOMETHING#3}`
1. Add Search Box and Search Results webparts to a page and connect them.
1. Edit the page, select the results webpart and edit properties
  1. Validate that the label of the result source property
  2. Validate the description below the result source property
  3. Validate that you can browse the built-in result sources in the combo box
  4. Validate that you can select one of the built-in result sources in the combo box, and that this affects search results
  5. Validate that you can specify the GUID of one of the custom result sources from step 1, 2 and 3, and that this affects search results
  6. Validate that you can specify the SPWeb|Source1, and that this is reflected in the source results
  7.  Validate that you can specify the SPSite|Source2, and that this is reflected in the source results
  8.  Validate that you can specify the SPSiteSubscription|Source3, and that this is reflected in the source results
  9. Validate that any other string values entered for the result source property are not persisted

### CODE REVIEW

**Comment by wobba** [comment]
@pettersh Can you please update the other locale files as well for the label.

File: search-parts/src/dataSources/SharePointSearchDataSource.ts [hunk 1]
```diff
@@ -948,19 +954,76 @@ export class SharePointSearchDataSource extends BaseDataSource<ISharePointSearch
...
+    /**
+     * Configures the SearchQuery to allow search by Result source Name.
+     * When searching by result source name, the Source name and level has to be set as properties
+     * More info here: https://www.techmikael.com/2015/01/how-to-query-using-result-source-name.html
+     * @param _resultSourceId the value from the properties
+     * @param searchQuery the SearchQuery being configured
+     */
+    private _setResultSourceByName(_resultSourceId: string, searchQuery: ISharePointSearchQuery): ISharePointSearchQuery {
+        const parts: string[] = _resultSourceId.split("|");
+        const level: string = parts[0];
```
Comment (wobba) [inline search-parts/src/dataSources/SharePointSearchDataSource.ts:1005]: Should we add a check to see if a | was included. That parts.length is 2.

**Comment by pettersh** [comment]
@wobba PR updated with translations from Google

**Comment by wobba** [comment]
We just merged in Norwegian translations - can you add this as well?

**Comment by pettersh** [comment]
@wobba Norwegian translations added

**Comment by wobba** [comment]
@FranckyC this one should be ok to merge now.

**Comment by FranckyC** [comment]
@obba Ok let me test it before ;)

**Pull request merged**